### PR TITLE
reuse common::streams::parquetsource

### LINKED ISSUE #2807: consider re-use `common::streams::ParquetSource` in fuse table
**Summary**

`ParquetSource` read parquet file in as DataBlock

https://github.com/datafuselabs/databend/blob/8b480244305cffe8e64e1d15c25091dc8214cb19/common/streams/src/sources/source_parquet.rs#L70-L74

It will be better to replace `block_reader::do_read`  with the `ParquetSource` in `FuseTable::do_read`
https://github.com/datafuselabs/databend/blob/f86c34b49d9e1620c1214eb3f6a3caffcd13f585/query/src/datasources/table/fuse/io/block_reader.rs#L59-L66

https://github.com/datafuselabs/databend/blob/12e4bb0f29b44873ffa3a48f6c0f3f346b7695f9/query/src/datasources/table/fuse/read.rs#L70-L72

### PR DESCRIPTION
**Pull request by Veeupup**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

reuse common::streams::parquetsource in fuse table read.
delete useless fuse::io::blcok_reader

## Changelog
- Improvement

## Related Issues

Fixes #2807 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by Veeupup** [comment]
I'm confused why this branch has conflicts with the master branch as I have deleted the file `query/src/datasources/table/fuse/io/block_reader.rs`. @dantengsky

**Comment by dantengsky** [comment]
there might be some modifications of block_reader in the main branch，before it is removed in this pr

**Comment by dantengsky** [comment]
pls take a look at this merged commit
https://github.com/datafuselabs/databend/pull/2839/commits/fd5efbf3bba43c8e329867e795c57ac4dfbcd4cb

**Comment by dantengsky** [comment]
> I forget to pull the newest commit….

that’s ok, never mind : )

File: query/src/datasources/table/fuse/read.rs [hunk 1]
```diff
@@ -66,10 +69,25 @@ impl FuseTable {
...
-        let stream = stream.then(move |part| {
-            io::do_read(part, da.clone(), projection.clone(), arrow_schema.clone())
-        });
+        let mut iter = futures::stream::iter(iter);
+        let stream = stream! {
+            while let Some(part) = iter.next().await {
+                let mut source = ParquetSource::new(
+                    da.clone(),
+                    part.name.clone(),
+                    Arc::new(DataSchema::from(arrow_schema.clone())),
```
Comment (dantengsky) [inline query/src/datasources/table/fuse/read.rs:78]: may be we could avoid some `clone`s here, by moving `Arc::new(DataSchema::from(arrow_schema.clone()))` outside, and only clone the `Arc<_>` inside?

Comment (Veeupup) [reply query/src/datasources/table/fuse/read.rs:78]: nice advice, I will fix it.

**Review by dantengsky — APPROVED** [review]
LGTM

**Pull request merged**